### PR TITLE
Complete dask demo

### PR DESCRIPTION
**Pull request by davidhassell**
Hi, not much to say here, other than I have address the commenting points discussed on Thursday. Also `cfdm` is no longer a dependency, as I now instantiate things in `demo.py` as 
```python
    # ----------------------------------------------------------------
    # Get the data as a lazy array with active capabilities
    # ----------------------------------------------------------------
    f = NetCDFArray(filename="file.nc", ncvar="q")

    # ----------------------------------------------------------------
    # Get the same data as an in-memory numpy array (with no active
    # capabilities)
    # ----------------------------------------------------------------
    nc = netCDF4.Dataset("file.nc", "r")
    x = nc.variables["q"][...]
    nc.close()

    # ----------------------------------------------------------------
    # Instantiate dask arrays from 'f' and 'x', each with the the same
    # arbitrary distribution of dask chunks.
    # ----------------------------------------------------------------
    dask_chunks = (3, 4)
    df = da.from_array(f, chunks=dask_chunks)
    dx = da.from_array(x, chunks=dask_chunks)
```

### CODE REVIEW

**Comment by valeriupredoi** [comment]
boss level, @davidhassell :beer: Did you want to add a test that runs the demo or some unit/integration tests? No probs if not, we can do that later :+1:

**Review by valeriupredoi — APPROVED** [review]

File: dask-demo/demo.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    be reinstated in an operational version.
+
+    """
+
+    def __init__(self, filename=None, ncvar=None):
+        self.filename = filename
+        self.ncvar = ncvar
+
+        nc = netCDF4.Dataset(self.filename, "r")
+        v = nc.variables[self.ncvar]
```
Comment (bnlawrence) [inline dask-demo/demo.py:27]: What happens if the keyword ncvar is not present? Would it be better to pass a NetCDF dataset (already instantiated) in along with a variable name (both required)?

Comment (bnlawrence) [reply dask-demo/demo.py:27]: (There would be some consequences. I would work through them, but you will have seen I have a git problem right now.)

Comment (davidhassell) [reply dask-demo/demo.py:27]: You'll get an exception with no `ncvar` or no `filename`. I don't think that it makes sense to pass in a `netCDF.Dataset` instance, as on principle we don't want those hanging around as open file handles, but we could make the `__init__` args positional only. I'll change that.

**Review by bnlawrence — APPROVED** [review]
I'm happy with this. @valeriupredoi Can you please merge.

**Comment by valeriupredoi** [comment]
cheers, gents! :beer:

**Pull request merged**